src/context/mod.rs

```diff
@@ -30,15 +30,14 @@ impl MessageContext {
         Ok(())
     }
 
-    pub fn format(&self, id: &str) -> Result<String, ContextError> {
-        match self.entries.get(id) {
-            Some(ref value) => {
-                match resolve(self, value) {
-                    Ok(msg) => Ok(msg),
-                    Err(_) => Err(ContextError::Generic),
-                }
-            }
-            None => Err(ContextError::Generic),
+    pub fn get_message(&self, id: &str) -> Option<&ast::Value> {
+        self.entries.get(id)
```

Comment:
I understand that right now `MessageContext` uses the runtime AST. I wonder if that's needed. Could we do without this special version of the AST in the Rust implementation?